include/deal.II/base/function_level_set.h

```diff
@@ -0,0 +1,110 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 - 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_function_level_set_h
+#define dealii_function_level_set_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/function.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Functions
+{
+  /**
+   * Signed-distance level set function of a sphere:
+   * $\psi(x) = \| x - x^c \| - R$.
+   * Here, $x^c$ is the center of the sphere and $R$ is its radius. This
+   * function is thus zero on the sphere, negative "inside" the ball having the
+   * sphere as its boundary, and positive in the rest of $\mathbb{R}^{dim}$.
+   *
+   * This function has gradient and Hessian equal to
+   * $\partial_i \psi(x) = (x - x^c)/\| x - x^c \|$,
+   * $\partial_i \partial_j \psi =
+   * \delta_{ij}/\| x - x^c \| - (x_i - x_i^c)(x_j - x_j^c)/\| x - x^c \|^3$,
+   * where $\delta_{ij}$ is the Kronecker delta function.
+   *
+   * @ingroup functions
+   */
+  template <int dim>
+  class SphereLevelSet : public Function<dim>
+  {
+  public:
+    /**
+     * Constructor, takes the center and radius of the sphere.
+     */
+    SphereLevelSet(const Point<dim> &center = Point<dim>(),
+                   const double      radius = 1);
+
+    double
+    value(const Point<dim> & point,
+          const unsigned int component = 0) const override;
+
+    Tensor<1, dim>
+    gradient(const Point<dim> & point,
+             const unsigned int component = 0) const override;
+
+    SymmetricTensor<2, dim>
+    hessian(const Point<dim> & point,
+            const unsigned int component = 0) const override;
```

Comment:
Given the singularity of this hessian at the center, can you add documentation to this function about what it returns at the center of the sphere? You can always copy the base class's documentation using `@copydoc` and then add another paragraph or `@note` about your specific case.

Reply:
It seems hard to return any consistent value when you are "floating-point-close" to the center. I've added a `@note`.